storybookjs/aem
# Feature/add aem metadata

Issue:

## What I did
Major goal: I added logic into a decorator and in the render function to handle the inclusion of component paths for use by the component loader. This removes the need for the user to include a components js file that would add all component paths to the window for consumption later.

Minor additions: 
- Pulled decorator options into the aem metadata decorator

#### This goes in the preview.js:
```
addDecorator(aemMetadata({
  componentIncludes: [
    require('../core/wcm/components/accordion/.content.xml'),
    require('../core/wcm/components/list/.content.xml'),
    require('../core/wcm/components/text/.content.xml'),
  ],
}));
```
#### However, this can be overwritten at the story config like this:
```
export default {
  title: 'Accordion',
  decorators: [
    aemMetadata({
      componentIncludes: [
        require('../core/wcm/components/accordion/.content.xml'),
        require('../core/wcm/components/list/.content.xml'),
        require('../core/wcm/components/text/.content.xml'),
      ],
    }),
  ]
};
```

#### In the same fashion, the decorator tag can be added globally:
```
addDecorator(aemMetadata({
      decorationTag: {
        cssClasses: ['text','component'],
        tagName: 'article'
      }
}));
```
#### or in the story config:
```
export default {
  title: 'Accordion',
  decorators: [
    aemMetadata({
     decorationTag: {
        cssClasses: ['text','component'],
        tagName: 'article'
      }
    })
  ]
};
```

## How to test
- run the app with the kitchen sink

If your answer is yes to any of these, please make sure to include it in your PR.
